app/models/user.rb

```diff
@@ -0,0 +1,10 @@
+class User < ApplicationRecord
+  before_save{self.email = email.downcase}
```

Comment:
```suggestion
  before_save{ email.downcase! }
```
so sánh giúp anh: email.downcase vs email.downcase! nhé

Reply:
email.downcase với email.downcase! đều là biến xâu ký tự thành chữ in thường.
Tuy nhiên, downcase! sẽ trả về nil nếu như xâu ký tự đấy không có gì thay đổi.
vd: a = "example"
a.downcase! = nil
a.downcase = example

Reply:
vẫn thiếu, e tìm hiểu thêm

Reply:
downcase không làm thay đổi chuỗi, còn downcase! thì có.
vd: a = "AB"
a.downcase! = "ab"
puts a => ab
a = "AB"
a.downcase = "ab"
puts a => AB